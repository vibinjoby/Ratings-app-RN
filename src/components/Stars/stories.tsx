import React from 'react'
import { storiesOf } from '@storybook/react-native'
import Stars from './component'

import Colors from '../../utilities/colors'

storiesOf('Components', module).add('Stars', () => (
  <Stars
    selectedColor={Colors.golden}
    selectable={false}
    selectedStars={3}
    onSelection={() => console.log('selected')}
    starHeight={15}
  />
))
