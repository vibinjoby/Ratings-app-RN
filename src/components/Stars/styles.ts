import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  container: {},
  starSpacing: { marginRight: 5 },
})
