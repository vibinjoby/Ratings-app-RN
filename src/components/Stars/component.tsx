import React from 'react'
import { FlatList, TouchableOpacity, View } from 'react-native'
import FontAwesome from 'react-native-vector-icons/FontAwesome'

import Colors from '../../utilities/colors'
import styles from './styles'

export interface StarsProps {
  selectable?: boolean
  selectedStars: number
  selectedColor?: string
  starHeight?: number
  spacing?: Record<string, unknown>
  onSelection?: (arg0: number) => void
  customStyle?: Record<string, unknown>
}

const Stars: React.FC<StarsProps> = ({
  selectable,
  selectedStars = 1,
  selectedColor,
  starHeight,
  spacing,
  onSelection,
}: StarsProps) => {
  return (
    <FlatList
      data={[1, 2, 3, 4, 5]}
      scrollEnabled={false}
      horizontal={true}
      renderItem={({ index }) => (
        <TouchableOpacity
          activeOpacity={1}
          onPress={() => selectable && onSelection && onSelection(index + 1)}
        >
          <FontAwesome
            testID={'font-icon'}
            name={selectedStars > index && selectedStars < index + 1 ? `star-half-full` : `star`}
            color={selectedStars > index && selectedStars ? selectedColor : Colors.unselectedStars}
            size={starHeight ? starHeight : 20}
          />
        </TouchableOpacity>
      )}
      ItemSeparatorComponent={() => <View style={[styles.starSpacing, spacing]} />}
      keyExtractor={(_, index) => index.toString()}
    />
  )
}

export default Stars
