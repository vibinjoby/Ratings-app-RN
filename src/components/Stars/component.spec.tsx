import 'react-native'
import React from 'react'
import Stars from './component'

import { render } from '@testing-library/react-native'
import Colors from '../../utilities/colors'

describe('Stars', () => {
  const mockFn = jest.fn()

  it('should render', () => {
    render(
      <Stars
        selectedColor={Colors.golden}
        selectable={false}
        selectedStars={3}
        onSelection={mockFn}
        starHeight={15}
      />,
    )
  })
})
