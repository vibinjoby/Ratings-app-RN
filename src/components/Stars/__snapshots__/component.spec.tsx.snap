// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars should render 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  data={
    Array [
      1,
      2,
      3,
      4,
      5,
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEnabled={false}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F5D547",
                "fontSize": 15,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="font-icon"
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginRight": 5,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F5D547",
                "fontSize": 15,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="font-icon"
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginRight": 5,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F5D547",
                "fontSize": 15,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="font-icon"
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginRight": 5,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#A5A6B7",
                "fontSize": 15,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="font-icon"
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginRight": 5,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#A5A6B7",
                "fontSize": 15,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="font-icon"
        >
          
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
